Source/OxyPlot/PlotModel/PlotModel.Rendering.cs

```diff
@@ -42,8 +42,11 @@ protected virtual void RenderOverride(IRenderContext rc, OxyRect rect)
         {
             lock (this.SyncRoot)
             {
+                var initialClipCount = 0;
+
                 try
                 {
+                    initialClipCount = rc.ClipCount;
```

Comment:
Does this need to be inside the try?

Reply:
Well we don't know how other `IRenderContext`s implement this property and if they might throw an exception (because of a faulty implementation).

I guess it is then a question whether we want to catch and render this exception, or throw it. I'd plead for catch and render, as we do with other exceptions during rendering.

Reply:
In that case, it might make sense to put a `try...catch` around the poping in the `catch`.

Reply:
Oh I see what you mean - maybe then it is better to move the assignment outside of the try... If `rc.ClipCount` doesn't work reliably, the exception message might not get properly rendered and it might be better to just throw.

Reply:
The problem is that throwing in `Render` has a habit of not showing up for the user to see (i.e. the control dies but the application doesn't crash and burn). That said, `LastException` will be set, so it's not totally undebuggable if something goes wrong, and you'd hope we can write reliable enough code that `ClipCount` isn't throwing regularly ;)